templates/common/baremetal/files/baremetal-mdns-publisher.yaml

```diff
@@ -24,6 +24,21 @@ contents:
         hostPath:
           path: "/etc/mdns"
       initContainers:
+      - name: verify-hostname
+        image: {{ .Images.baremetalRuntimeCfgImage }}
+        env:
+          - name: DEFAULT_LOCAL_HOSTNAME
+            value: "localhost"
+        command:
+        - "/bin/bash"
+        - "-c"
+        - |
+          #/bin/bash
+          while [ $(hostname) == "$DEFAULT_LOCAL_HOSTNAME" ]
```

Comment:
Maybe want to put quotes around $(hostname)? I don't know if there would be anything in a hostname that would mess up the check, but this being bash I generally err on the side of caution.

Reply:
@cybertron - Done. 10x!